sdkproject/Assets/Mapbox/Examples/Scripts/QuadTreeCameraMovement.cs

```diff
@@ -0,0 +1,134 @@
+﻿namespace Mapbox.Examples
+{
+	using Mapbox.Unity.Map;
+	using Mapbox.Unity.Utilities;
+	using Mapbox.Utils;
+	using UnityEngine;
+	using System;
+
+	public class QuadTreeCameraMovement : MonoBehaviour
+	{
+		[SerializeField]
+		[HideInInspector]
+		public float _zoomSpeed = 50f;
+
+		[SerializeField]
+		//[HideInInspector]
+		public Camera _referenceCamera;
+
+		[SerializeField]
+		QuadTreeTileProvider _quadTreeTileProvider;
+
+		[SerializeField]
+		AbstractMap _dynamicZoomMap;
+
+		private Vector3 _origin;
+		Vector3 _mousePosition;
+		Vector3 _mousePositionPrevious;
+		bool _shouldDrag;
+		private Transform _originalCameraPosition;
+
+		/// <summary>min of y range camera is allowed to move in</summary>
+		private int _cameraZoomingRangeMinY;
+		public int CameraZoomingRangeMinY { get { return _cameraZoomingRangeMinY; } }
+		/// <summary>max of y range camera is allowed to move in</summary>
+		private int _cameraZoomingRangeMaxY;
+
+		public int CameraZoomingRangeMaxY { get { return _cameraZoomingRangeMaxY; } }
+
+		void Start()
+		{
+			if (null == _referenceCamera)
+			{
+				_referenceCamera = GetComponent<Camera>();
+				if (null == _referenceCamera) { Debug.LogErrorFormat("{0}: reference camera not set", this.GetType().Name); }
+			}
+
+			//put camera facing down. 
+			//_referenceCamera.transform.rotation = new Quaternion(0.7f, 0, 0, 0.7f);
```

Comment:
For this example it's essential that the camera looks down straight.
I think we should enforce it.
When rotating camera around X-axis the whole effect blows up.

Reply:
Camera looking straight down is no longer a requirement. Map is on the ground, and only the part of the map seen in the camera's viewport is drawn.

Reply:
Nice! 🎉